### PR TITLE
Fix FromPrimitive overflow check

### PR DESCRIPTION
**Pull request by kaidokert**
## Summary
- handle overflow in `FromPrimitive::from_u64`
- update `NumCast` to use the checked conversion
- test overflow behavior for `from_u64`

## Testing
- `cargo test --quiet`

------
https://chatgpt.com/codex/tasks/task_e_687dcdd370dc8331a31d7e9872b5c88d

## Summary by Sourcery

Prevent unchecked overflow when converting from u64 to FixedUInt by adding range checks and corresponding tests.

Enhancements:
- Add overflow check in FromPrimitive::from_u64 to return None for out-of-range inputs.

Tests:
- Add tests for from_u64 overflow behavior on 8-bit and 16-bit FixedUInt types.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of numeric conversions to prevent overflow when converting large values to fixed-size integers.

* **Tests**
  * Added tests to verify correct behavior when converting values that exceed the capacity of fixed-size integers.

### CODE REVIEW

File: src/fixeduint/num_traits_casts.rs [hunk 1]
```diff
@@ -30,6 +30,11 @@ impl<T: MachineWord, const N: usize> num_traits::FromPrimitive for FixedUInt<T,
         None
     }
     fn from_u64(input: u64) -> Option<Self> {
+        if let Some(max) = Self::max_value().to_u64() {
+            if input > max {
+                return None;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/num_traits_casts.rs:40]: nah

File: src/fixeduint/num_traits_casts.rs [hunk 2]
```diff
@@ -30,6 +30,11 @@ impl<T: MachineWord, const N: usize> num_traits::FromPrimitive for FixedUInt<T,
         None
     }
     fn from_u64(input: u64) -> Option<Self> {
+        if let Some(max) = Self::max_value().to_u64() {
+            if input > max {
+                return None;
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/num_traits_casts.rs:38]: fixed now
